app/api/deps/core.py

```
@@ -14,3 +15,27 @@ async def get_supabase_client():
+async def get_current_user(
+    credentials: HTTPAuthorizationCredentials = Depends(security),
+    supabase: AsyncClient = Depends(get_supabase_client),
+):
+    token = credentials.credentials
+    try:
+        response = await supabase.auth.get_user(token)
+
+        if not response:
+            raise
+
+        return response.user
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)